### ISSUE TITLE
Using python3.10 feature but python3.9 is specified in README.md (FABulous2.0-development branch)

### ISSUE THREAD

**Issue by IAmMarcelJung** [FPGA-Research-Manchester/FABulous#228]
When I wanted to test the changes in #227, I saw that the python version to use is specified as `python3.9` in the `FABulous2.0-development`  branch. Therefore I set my venv to `python3.9` and set it up as usual. When I wanted to create the project, using `FABulous -c demo` I got the following error (shortened):

```console
File "FABulous/fabric_definition/Fabric.py", line 249, in Fabric
    def getTileByName(self, name: str) -> Tile | None:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```
I then found out that the `union` type was introduced in [PEP 604](https://peps.python.org/pep-0604/) and is only supported in `python3.10` and higher. Therefore I suggest we either remove the union type or change the suggested python version in `README.md` to at least `python3.10` in the `FABulous2.0-development` branch (which should be preferred IMHO).

If there is an error on my side please let me know, but I think the PEP is quite clear.

**Comment by KelvinChung2000** [comment]
We might as well move to python3.12 since our CI is currently running at python3.12.

**Comment by IAmMarcelJung** [comment]
Yeah, then we should specify `python3.12` in `README.md` and also in `pyproject.toml` (thanks to @EverythingElseWasAlreadyTaken for the hint). I will create a PR for this.

**Comment by IAmMarcelJung** [comment]
> We might as well move to python3.12 since our CI is currently running at python3.12.

For me it looks like we still use [3.10 in the CI](
https://github.com/FPGA-Research-Manchester/FABulous/blob/FABulous2.0-development/.github/workflows/fabric_gen.yml#L13-L16). Or am I looking at the wrong file? Just want to make sure before I change the docs to 3.12 :)

**Comment by KelvinChung2000** [comment]
I thought it is on 3.12🤔. Anyway, I think we should still move to 3.12.

________________________________

**Comment by IAmMarcelJung** [comment]
No problem! So should I change everything to 3.12? Also for the docs and the CI?

**Comment by KelvinChung2000** [comment]
Yes, please.
________________________________
